plugins/generic/acron/PKPAcronPlugin.inc.php

```diff
@@ -31,38 +36,34 @@
 
 class PKPAcronPlugin extends GenericPlugin
 {
-    /** @var string $_workingDir */
-    public $_workingDir;
-
-    /** @var array $_tasksToRun */
-    public $_tasksToRun;
+    private array $_tasksToRun;
 
     /**
      * @copydoc Plugin::register()
      *
      * @param null|mixed $mainContextId
      */
-    public function register($category, $path, $mainContextId = null)
+    public function register($category, $path, $mainContextId = null): bool
     {
         $success = parent::register($category, $path, $mainContextId);
-        HookRegistry::register('Installer::postInstall', [&$this, 'callbackPostInstall']);
+        HookRegistry::register('Installer::postInstall', fn () => $this->_callbackPostInstall(...func_get_args()));
```

Comment:
Hmm, this syntax feels less friendly to me than the array callback function, and it wraps the actual callback in an extra function call. Are there advantages to this?

Reply:
Ah, it's mostly personal bias against using strings for things that should be typed. PHP should allows us to reference first-class functions without using strings.

Benefits: a code editor is able to show invalid method calls (e.g. inexistent method, bad arguments, non-public), it must also make the life of code analyzers easier.

Reply:
We'll get better options for [first class callable syntax](https://www.php.net/manual/en/functions.first_class_callable_syntax.php) when we require a baseline PHP 8.1:
```php
HookRegistry::register('Installer::postInstall', $this->callbackPostInstall(...));
```
I'd suggest we wait for that; I agree that having code completion is much better than the old array construction but the semantics of another wrapper function and not-totally-obvious `...func_get_args()` invocation muddy the waters.

Reply:
Agreed, let's wait for 8.1 😁
I just updated the PR again.